File: VB/Scaffolding.ValidationErrors/ViewModels/DepartmentViewModel.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Linq
Imports System.Linq.Expressions
Imports System.Collections.Generic
Imports System.Collections.ObjectModel
Imports DevExpress.Mvvm
Imports DevExpress.Mvvm.POCO
Imports Scaffolding.ValidationErrors.Common.Utils
Imports Scaffolding.ValidationErrors.DepartmentContextDataModel
Imports Scaffolding.ValidationErrors.Common.DataModel
Imports Scaffolding.ValidationErrors.Model
Imports Scaffolding.ValidationErrors.Common.ViewModel

Namespace Scaffolding.ValidationErrors.ViewModels
	Partial Public Class DepartmentViewModel
		Inherits SingleObjectViewModel(Of Department, Integer, IDepartmentContextUnitOfWork)
		Public Sub New()
			Me.New(UnitOfWorkSource.GetUnitOfWorkFactory())
		End Sub
		Public Sub New(ByVal unitOfWorkFactory As IUnitOfWorkFactory(Of IDepartmentContextUnitOfWork))
			MyBase.New(unitOfWorkFactory, Function(x) x.Departments)
		End Sub
	End Class
End Namespace

File: CS/Scaffolding.ValidationErrors/ViewModels/DepartmentViewModel.cs
using System;
using System.Linq;
using System.Linq.Expressions;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using DevExpress.Mvvm;
using DevExpress.Mvvm.POCO;
using Scaffolding.ValidationErrors.Common.Utils;
using Scaffolding.ValidationErrors.DepartmentContextDataModel;
using Scaffolding.ValidationErrors.Common.DataModel;
using Scaffolding.ValidationErrors.Model;
using Scaffolding.ValidationErrors.Common.ViewModel;

namespace Scaffolding.ValidationErrors.ViewModels {
    public partial class DepartmentViewModel : SingleObjectViewModel<Department, int, IDepartmentContextUnitOfWork> {
        public DepartmentViewModel()
            : this(UnitOfWorkSource.GetUnitOfWorkFactory()) {
        }
        public DepartmentViewModel(IUnitOfWorkFactory<IDepartmentContextUnitOfWork> unitOfWorkFactory)
            : base(unitOfWorkFactory, x => x.Departments) {
        }
    }
}
